### PR TITLE
DEP Upgrade webonyx/graphql-php to v14

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10250

### CODE REVIEW

File: tests/Fake/FakeResolveInfo.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace SilverStripe\GraphQL\Tests\Fake;
+
+use GraphQL\Type\Definition\FieldDefinition;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Schema;
+
+class FakeResolveInfo extends ResolveInfo
```
Comment (GuySartorelli) [inline tests/Fake/FakeResolveInfo.php:11]:
```suggestion
class FakeResolveInfo extends ResolveInfo implements TestOnly
```
Also add `use` statement.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Remove the `use GraphQL\Type\Definition\ResolveInfo;` statements from files where you've replaced that with the new `FakeResolveInfo` class.

Why do we need a `FakeResolveInfo` class? I just want to be sure that it's okay to do this for tests without getting false positive results.

In general looks good (haven't tested it locally) - couple of small changes and a bit of context to be given there.

File: tests/Fake/FakeResolveInfo.php (hunk 1 above)
Comment (emteknetnz) [reply tests/Fake/FakeResolveInfo.php:11]: > Why do we need a FakeResolveInfo class? I just want to be sure that it's okay to do this for tests without getting false positive results.

Between v0.12 and v14 of webonyx/graphql-php the constructor for ResolveInfo changed from an array, to 10 params that are mostly non-nullable.  For the majority of the tests we don't even need the ResolveInfo object for anything, it's just a non-optional argument.  So the FakeResolveInfo is just a nice way to fill in the 10 constructor arguments really.

Asset-admin, versioned and elemental already have existing FakeResolveInfo classes so I just copied what was done there.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Missed a `use GraphQL\Type\Definition\ResolveInfo;` in `tests/Scaffolding/Scaffolders/CRUD/ReadTest.php`

**Review by GuySartorelli — APPROVED** [review]
I'm happy with this.
MOG.

**Pull request merged**